types/ramda/index.d.ts

```diff
@@ -807,78 +810,28 @@ export function contains<T>(a: T): (list: readonly T[]) => boolean;
  * are passed as arguments to the converging function to produce the return value.
  */
 export function converge<
-    TArgs extends any[],
     TResult,
-    R1,
-    R2,
-    R3,
-    R4,
-    R5,
-    R6,
-    R7,
-    RestFunctions extends Array<(...args: TArgs) => any>,
+    FunctionsList extends ReadonlyArray<Fn> &
+        IfFunctionsArgumentsDoNotOverlap<_Fns, 'Functions arguments types must overlap'>,
```

Comment:
I wish we could handle this, without this way of producing TypeScript. Especially since the case being handled is quite basic TypeScript Error stuff. But I also don't really know how to improve that, so I guess its fine.